### PR TITLE
[#1235] ported domain example to use java instead of xtend

### PR DESCRIPTION
**Pull request by cdietrich**
[#1235] ported domain example to use java instead of xtend
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+		append(builder, property.getType());
+		return style(builder.toString());
+	}
+
+	public Object text(Operation operation) {
+		StringBuilder builder = new StringBuilder();
+		builder.append(Strings.notNull(operation.getName()));
+		builder.append("(");
+		boolean isFirst = true;
+		EList<JvmFormalParameter> _params = operation.getParams();
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java]: `params` is a better name

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/serializer/DomainmodelSyntacticSequencer.java [hunk 2]
```diff
@@ -9,16 +9,14 @@
...
 public class DomainmodelSyntacticSequencer extends AbstractDomainmodelSyntacticSequencer {
-  @Override
-  protected String getArrayBracketsToken(final EObject semanticObject, final RuleCall ruleCall, final INode node) {
-    if ((node != null)) {
-      return this.getTokenText(node);
-    }
-    return "[]";
-  }
+	@Override
+	protected String getArrayBracketsToken(final EObject semanticObject, final RuleCall ruleCall, final INode node) {
```
Comment (tivervac) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/serializer/DomainmodelSyntacticSequencer.java]: No need for final

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+			append(builder, type);
+			EList<JvmTypeReference> arguments = ((JvmParameterizedTypeReference) typeRef).getArguments();
+			if (!arguments.isEmpty()) {
+				builder.append("<");
+				Iterator<JvmTypeReference> iterator = arguments.iterator();
+				while (iterator.hasNext()) {
+					JvmTypeReference jvmTypeReference = iterator.next();
+					append(builder, jvmTypeReference);
+					if (iterator.hasNext()) {
+						builder.append(",");
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java]: ", " <- missing space after comma

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+			}
+			isFirst = false;
+			append(builder, param.getParameterType());
+		}
+		builder.append(") : ");
+		append(builder, operation.getType());
+		return style(builder.toString());
+	}
+
+	protected void append(StringBuilder builder, JvmTypeReference typeRef) {
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java]: This entire method looks like `builder.append(typeRef.getSimpleName())`

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+				});
+			}
+		});
+	}
+
+	@Check
+	public void checkOperationNamesAreUnique(Entity entity) {
+		JvmGenericType inferredJavaClass = IterableExtensions
+				.head(Iterables.filter(jvmModelAssociations.getJvmElements(entity), JvmGenericType.class));
+		domainmodelJvmModelHelper.handleDuplicateJvmOperations(inferredJavaClass, (Collection<JvmOperation> jvmOperations) -> {
```
Comment (szarnekow) [inline org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.java]: Lambda parameter type can be left out

File: org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java (hunk 3 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel.ui/src/org/eclipse/xtext/example/domainmodel/ui/labeling/DomainmodelLabelProvider.java]: comma not there in the original one

**Pull request merged**